diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6fc039e..6eacc12 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,7 +59,7 @@ if(WIN32 OR MINGW)
 endif()
 
 add_library(hiredis_cluster
-  SHARED
+  STATIC
   ${hiredis_cluster_sources})
 
 if(NOT MSVC)
@@ -165,7 +165,7 @@ if(ENABLE_SSL)
   endif()
 
   add_library(hiredis_cluster_ssl
-    SHARED hircluster_ssl.c)
+    STATIC hircluster_ssl.c)
   set_target_properties(hiredis_cluster_ssl
     PROPERTIES VERSION "${HIREDIS_CLUSTER_SONAME}")
   target_link_libraries(hiredis_cluster_ssl
